contract.md

```diff
@@ -2,39 +2,124 @@
+than a completely new language, and additionally, it could use syntax that has
+been tested and reviewed by millions over many years which would help ensure
+that it is usable for a wide audience.
+
+In terms of the skills I develop, I primarily want to better hone my ability to
+make a syntax that is intuitive and concise.
+Any language, no matter how powerful, could be effectively useless if the syntax
+isn't something that users can understand and enjoy using.
+Thus, to me this seems like the most important skill to develop in terms of
+making DSLs.
+In addition to developing clean syntax, I hope to improve my ability to convert
+a syntax to my desired output in a straightforward and elegant way.
+I really enjoyed this part of the previous assignment and I hope to explore more
+ways of interpreting structurally complex syntaxes that I may create.
 
 ## Concepts / skills
 
-_What concepts and / or skills from class would you like to work on, as part of your
-project? Are there any concepts / skills that we haven't covered in class that you would
-like to work on, as part of your project (for example, other things related to DSLs that
-you have come across, or topics from other classes)? Why are you interested in working on
-these things?_
+Two things I especially want to work on are parsing unique syntaxes and
+interpreting the parsed syntax into the desired output.
+For parsing, I enjoyed working with the basic syntaxes we developed for the
+previous homework, but I am curious what additional functionality I will have to
+learn for this larger project with a syntax that will continually grow and
+change as I add and modify design features.
+Additionally, since I don't plan to use Scala as my host language, I will be
+exploring how parsing works in other languages and am curious to see the
+differences.
 
-## Time management plan
+For interpreting, I found it difficult, but rewarding to find an intermediate
+representation that best fits the output of my program.
+For this project, I plan to have output in multiple languages (HTML, CSS, and
+JS) which are all related, but definitely distinct.
+This could definitely make the job of interpreting more complicated and make it
+more difficult to find an IR that is well suited for converting parts to all
+three of these languages, but I am excited for the challenge.
 
-_How do you plan to set aside time outside of class, to work on the project? Are there
-intermediate milestones that you can create, to help you make consistent progress?_
+## Time management plan
 
-## Teamwork plan
+Time management is one of my weaknesses, so this may be a harder part of the
+assignment.
+Luckily, this semester I don't have any other major assignments due on Mondays
+which means I can be focused on this project on the weekends.
+Although I may not always be able to get everything done by midnight, I always
+have Monday morning and will make sure to get what I want done before class.
```

Comment:
I ask believe it's really important to have stuff prepared to discuss in class. This will help you get more input on your work.